### DISCUSSION TITLE
Nuget package versions

### DISCUSSION THREAD

**Discussion by SrinathSundaresan** [microsoft/Dynamics365Commerce.Solutions#30] [Q&A] [answered]
How the versioning of NuGet packages is done relative to the CSU version. Can the NuGet packages be on a higher version and are the lower version backward compatible.

**Comment by SrinathSundaresan** [comment]
@madyke Posting here as discussed

**Comment by madyke** [comment, marked as answer]
Hi @SrinathSundaresan - the NuGet packages for client apps like Store Commerce and the Commerce SDK must be the **same or older** than the version of CSU. That means CSU will be backwards compatible with older versions of client apps. So if CSU is on version 10.0.30, then Store Commerce must be on version 10.0.30 or earlier (such as 10.0.29 or 10.0.28).

You can find more info in this article that provides an overview of the versioning requirements and dependencies for all D365 Commerce components, if you need it:
[Dynamics 365 Commerce component versioning requirements](https://learn.microsoft.com/en-us/dynamics365/commerce/arch-component-versioning)

**Comment by SrinathSundaresan** [comment]
@madyke Are NuGet package majorversions.* compatible for a csu release?

**Reply by madyke** [reply]
Hi @SrinathSundaresan - no, but Major.Minor.* versions will be compatible. So for example, if the customer has CSU version 10.0.32, then Commerce SDK NuGet packages that begin with 9.42 would all be compatible (as well as older versions like I described above, such as 9.41 and 9.40).

You can find more info about package versioning and examples of currently supported versions in [this table](https://learn.microsoft.com/en-us/dynamics365/commerce/arch-component-versioning#current-supported-versions).
